src/Vault.sol

```diff
@@ -374,16 +374,26 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
    * @inheritdoc ERC4626
    * @dev We use type(uint96).max cause this is the type used to store balances in TwabController.
    */
-  function maxDeposit(address) public view virtual override returns (uint256) {
-    return _isVaultCollateralized() ? type(uint96).max : 0;
+  function maxDeposit(address recipient) public view virtual override returns (uint256) {
+    if (!_isVaultCollateralized()) return 0;
+
+    uint256 _vaultMaxDeposit = type(uint96).max - _convertToAssets(balanceOf(recipient), Math.Rounding.Down);
```

Comment:
Wouldn't this be rounded up? We don't want any portion to overflow

Reply:
Ah yes, good point.
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/11/commits/b7fcb005cf655810875ba4b85345ee5b2301603e